README.md

```diff
@@ -1 +1,571 @@
+  isVerified: Boolean!
+  verifiedAt: ISO8601DateTime
+}
+```
+### Interfaces and Unions
+
+Unions are semantically just marker interfaces (interfaces with zero fields) so I will refer to both interfaces and unions as interfaces.
+
+When choosing field types, prefer interfaces instead of concrete types.
+
+Instead of a `Comment` type, consider a `Comment` interface and a `PostComment` type. Instead of a `Message` type, consider a `Message` interface and a `ChatMessage` type.
+
+### References
+
+Refer to objects, not IDs. GraphQL allows direct traversal so let clients do that.
+
+```graphql
+# good
+type User {
+  invitedBy: User
+}
+
+# bad
+type User {
+  invitedBy: ID
+}
+```
+
+### Relay
+
+Implement the Relay spec. Read [this](https://relay.dev/docs/en/graphql-server-specification).
+
+One thing often neglected in Relay's cursor pagination is that there is nothing preventing the extension of connections, edges, and page info with additional fields. Cursor pagination can get unwieldy with nested associations. However, this can be avoided by treating your data model as nodes and edges, instead of only using edges for pagination. For example, organization memberships are conceptually edges between users and organizations and should be able to be queried as such.
+
+### Current User
+
+It is very common to provide the current user with more information about themselves and their current session than anyone else can access. This makes it problematic to give the current user direct access to their `User` instance. Instead, create a type specifically for this. Typically, this type is called `Viewer` or `Me`. Add a query field that returns the singleton instance of that type.
+
+```graphql
+type Viewer {
+  id: ID!
+  name: String
+}
+
+type Query {
+  # ...
+  viewer: Viewer
+}
+```
+
+Alternatively, these fields can live on the query root, but that is messy and, historically, Relay client required that those fields not be root fields. Therefore, `Viewer` or `Me` is standard.
+
+
+### Arguments
+
+Use field arguments instead of almost identical fields.
+
+It is common to need to provide the client with multiple things that are almost, but not quite, identical. For example, the client might need to access the viewer's avatar in multiple sizes. Instead of adding `avatarSmall`, `avatarMedium`, etc. fields, consider a schema like this:
+
+```graphql
+type User {
+  avatarURL(input: UserAvatarInput!): URL
+}
+
+input UserAvatarInput {
+  width: Int!
+  height: Int!
+}
+```
+
+Fields are essentially named-parameter functions. Even though they can take any number of arguments, they should be nullary or unary; they should not take more than one argument. If multiple arguments will be needed, the field should take a single `input` argument that is an input object named `[TypeNameFieldName]Input`. It is best to default to always doing this input object when arguments are needed and only consider scalar input in exceptional circumstances.
+
```

Comment:
Maybe list these best practices under `i)` and `ii)`, in order to see the delineation.
```
i) Use field arguments instead of...
...
ii) Fields are essentially named-...
...
```